[quant] ConvPackedParams with TorchBind

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#35923 [quant] ConvPackedParams with TorchBind**

Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

jerryzh168 committed:
[quant] ConvPackedParams with TorchBind

Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[WIP][quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[WIP][quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[WIP][quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@jamesr66a do I need to change anything about ONNX?

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@pytorchbot retest this please

jamesr66a commented on aten/src/ATen/native/quantized/cpu/fbgemm_utils.h:
```diff
@@ -32,14 +33,96 @@ struct CAFFE2_API PackedLinearWeightFp16 {
+      at::Tensor input,
+      double output_scale,
+      int64_t output_zero_point) override {
+    return apply_impl<false>(input, output_scale, output_zero_point);
```
Question: will this `apply_impl` callsite cause instantiation in compilation units other than `qconv.cpp`? I see you've instantiated `apply_imp` in `qconv.cpp`, but should the definitions of `apply` and `apply_relu` be in that file as well?

jerryzh168 replied:
right, I think these can be removed

jerryzh168 replied:
looks like we can remove all explicit template instantiations in qconv.cpp

jerryzh168 replied:
Looks like we still need them

jerryzh168 replied:
moved these to qconv.cpp

jamesr66a commented on aten/src/ATen/quantized/Quantizer.cpp:
```diff
@@ -70,18 +70,8 @@ void checkZeroPoint(std::string fn_name, int64_t zero_point) {
-                "zero_point",
-                i,
-                "is out of range.");
+    checkZeroPoint<T>(fn_name, zero_points[i].item<int64_t>());
```
Is this related to the rest of the changes?

jerryzh168 replied:
no, will revert

jamesr66a commented on test/backward_compatibility/check_backward_compatibility.py:
```diff
@@ -25,6 +25,17 @@
+    ('quantized::conv3d_prepack', datetime.date(2020, 4, 15)),
+    ('quantized::conv3d_unpack', datetime.date(2020, 4, 15)),
+    ('quantized::conv3d', datetime.date(2020, 4, 15)),
+    ('quantized::conv3d_relu', datetime.date(2020, 4, 15)),
```
I think we're also going to have to figure out if this will break serialized TorchScript modules. The way that would happen is if the packed parameter attributes in the serialized Module are typed as Tensor, rather than Conv*dPackedParamsBase. Can you jit.save() a model containing one of these modules before this change, load it in after this change, and confirm if this is indeed a breaking situation?

jerryzh168 replied:
is `torch.jit.save` calling `__getstate__` and `__setstate__` function? I didn't change these functions so it should still work(we are still saving unpacked weights directly)

jerryzh168 replied:
Yes, this will break serialized torchscript modules since the whole module is saved with `torch.jit.save`:
```
import torch
class M(torch.nn.Module):
    def __init__(self):
        super(M, self).__init__()
        self.conv = torch.nn.quantized.Conv2d(3, 3, 3)

    def forward(self, x):
        return self.conv(x)

a = torch.jit.script(M())
x = torch.quantize_per_tensor(torch.rand(1, 3, 10, 10), 0.5, 5, torch.quint8)
b = torch.jit.trace(M(), x)
print(a)
print(b)
# torch.jit.save(a, "a.pt")
# torch.jit.save(b, "b.pt")

a = torch.jit.load("a.pt")
b = torch.jit.load("b.pt")
print(a, b)
```

Error:
```
Traceback (most recent call last):
  File "c.py", line 18, in <module>
    a = torch.jit.load("a.pt")
  File "/data/users/jerryzh/anaconda3/envs/py3/lib/python3.7/site-packages/torch/jit/__init__.py", line 227, in load
    cpp_module = torch._C.import_ir_module(cu, f, map_location, _extra_files)
RuntimeError:

quantized::conv2d(Tensor _0, __torch__.torch.classes.quantized.Conv2dPackedParamsBase _1, float _2, int _3) -> (Tensor _0):
Expected a value of type '__torch__.torch.classes.quantized.Conv2dPackedParamsBase' for argument '_1' but instead found type 'Tensor'.
:
  File "/data/users/jerryzh/anaconda3/envs/py3/lib/python3.7/site-packages/torch/nn/quantized/modules/conv.py", line 337
        if len(input.shape) != 4:
            raise ValueError("Input shape must be `(N, C, H, W)`!")
        return ops.quantized.conv2d(
               ~~~~~~~~~~~~~~~~~~~~ <--- HERE
            input, self._packed_params, self.stride, self.padding,
            self.dilation, self.groups, self.scale, self.zero_point)
Serialized   File "code/__torch__/torch/nn/quantized/modules/conv.py", line 36
    _13 = [_11, _12]
    _14, _15, = _4
    _16 = ops.quantized.conv2d(input, _1, _10, _13, [_14, _15], _5, _6, _7)
          ~~~~~~~~~~~~~~~~~~~~ <--- HERE
    return _16
  def __getstate__(self: __torch__.torch.nn.quantized.modules.conv.Conv2d) -> Tuple[int, int, Tuple[int, int], Tuple[int, int], Tuple[int,
int], Tuple[int, int], bool, int, int, str, Tensor, Optional[Tensor], float, int, bool]:
```

jamesr66a replied:
@jerryzh168 right, the issue here is that the ScriptModules are serialized with the packed parameters annotated as type Tensor. You can probably fix this by replicating what I've done on the import path: https://github.com/pytorch/pytorch/pull/34140/files#diff-93507b8ace0ea4890fdafb4f18e55f46R265

jerryzh168 replied:
BC is fixed now for `torch.jit.save`

jerryzh168 replied:
For documentation purpose, here is what we do(paste from chat): 
1) two overloads of conv2d, one with the old schema and one with the new. the one with the old schema throws an exception and says to use the new one
2) a pass that converts the old conv2d callsites to the new schema on ScriptModules after load()

jamesr66a reviewed (commented):
Nice! I think the code looks good, but we're going to have to make sure this is solid from a compatibility standpoint. A few things that come to mind are:

- ONNX export. I had to do some shuffling in my PR to get that to work. The CI jobs should indicate whether this is needed or not
- ScriptModule BC. I added a comment inline in `test_backward_compatibility.py` about this
- `__torch_function__` - dunno if this applies to Conv, but I ran into it in the RNN PR and I haven't figured out how to fix it yet

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 commented on test/quantization/test_quantized_nn_mods.py:
```diff
@@ -567,16 +567,29 @@ def _test_conv_api_impl(
+        # Note that _use_new_zipfile_serialization=False doesn't work
+        # TODO: remove _use_new_zipfile_serialization=True
+        # after https://github.com/pytorch/pytorch/pull/32958 is landed
+        torch.save(model_dict, bytes_io, _use_new_zipfile_serialization=True)
```
@raghuramank100

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jamesr66a reviewed (commented):
Seems reasonable, but internal builds are showing some BC breakage

jerryzh168 commented on aten/src/ATen/native/quantized/cpu/conv_packed_params.h:
```diff
@@ -0,0 +1,23 @@
+template <int kSpatialDim = 2>
+struct ConvPackedParamsBase : public torch::jit::CustomClassHolder {
+  virtual at::Tensor apply(
+      at::Tensor input,
```
sure, but it should be similar since Tensor is a pointer

jerryzh168 commented on aten/src/ATen/native/quantized/cpu/fbgemm_utils.cpp:
```diff
@@ -197,3 +204,131 @@ Tensor ConvertToChannelsLast3dTensor(const Tensor& src) {
+      torch::List<at::Tensor>,
+      torch::List<at::Tensor>,
+      torch::List<at::Tensor>,
+      at::Tensor>;
```
right, the next bc breaking change would be change from torch::List<at::Tensor> to torch::List<int64_t> when onnx is fixed, do you know how we can make backward compatible changes at that point? I feel it might be easier to separate the arguments instead of putting them together?

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 commented on aten/src/ATen/native/quantized/cpu/qconv.cpp:
```diff
@@ -169,485 +699,54 @@ class QConvInt8 final {
+    TORCH_CHECK(false, "Arguments [stride, padding, dilation, groups] in \
+     ops.quantized.conv" + c10::to_string(kSpatialDim) + "d, " +
+     "ops.quantized.conv" + c10::to_string(kSpatialDim) + "d_relu, \
+     have been removed, please update your model to remove these arguments.");
```
this happens when user calls torch.ops.quantized.conv2d directly or in calling quantized functional ops in the serialized code.

jerryzh168 replied:
I don't think this can be runnable, don't we need to modify the type as we do in `qconvAttributeAssignmentSpecialHandlingHack` in order to make it work?

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jamesr66a reviewed (approved):
Great! Land if there are no more FBCode failures

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] ConvPackedParams with TorchBind"


Summary:

Test Plan:
tbd

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20957089](https://our.internmc.facebook.com/intern/diff/D20957089)

[ghstack-poisoned]
